app/main.py

```diff
@@ -1,13 +1,27 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return dict(
+        line=error["line_number"],
+        column=error["column_number"],
+        message=error["text"],
+        name=error["code"],
+        source="flake8"
+    )
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(one) for one in errors],
+        "path": file_path,
+        "status": "failed" if errors != [] else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
```

Comment:
use the existing format_single_linter_file function to avoid code duplication.

Reply:
Fixed code duplication